code/game/machinery/bots/floorbot.dm

```diff
@@ -705,15 +717,27 @@ var/global/list/floorbot_targets=list()
 	return
 
 
-/obj/item/weapon/storage/toolbox/mechanical/attackby(var/obj/item/stack/tile/plasteel/T, mob/user as mob)
+/obj/item/weapon/storage/toolbox/attackby(var/obj/item/stack/tile/plasteel/T, mob/user as mob)
 	if(!istype(T, /obj/item/stack/tile/plasteel) || src.contents.len >= 1) //Only do this if the thing is empty
 		return ..()
+	var/list/allowed_types = list(
```

Comment:
Not very elegant. You could make a proc that returns nothing on the base case, and the skin identifier for the three colored toolboxes. You'd use it to get rid of the `istype(src)`s below as well.